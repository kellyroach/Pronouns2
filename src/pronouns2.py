################################################################
#
#    pronouns2.py
#
################################################################

"""Main project Python module to import which imports everything else."""

import argparse
from doc import *
from find_files import *
from varying_rho import *

################################################################
#
#     Module Dependencies
#
# pronouns2
#   find_files
#   varying_rho
#     doc
#       demo
#         examples
#         table_interp
#           view
#             lexicon
#               node_proc
#                 trc
#                   globals
#         table_proc
#           abstract
#             modern
#               tex
#               view
#           chaining
#             modern
#             spread
#               trc
#           latex
#             modern
#           parse_proc
#             lexicon
#           secondary_uty
#             primary_uty
#               trc
#             modern
#             lexicon
#           txt
#             view
#
# NOTE: Except for trc.py, any module which must depend on
# globals.py should indirectly depend on globals.py through
# dependence on some other Python module or trc.py as the
# last resort.  Our strict policy keeps the tree simpler.
################################################################

def main():
    """Main entry point for the pronouns2.py command-line tool.
    
    Supports command line formats:
    python pronouns2.py
    python pronouns2.py "10.1"
    python pronouns2.py --file="MyFile"
    python pronouns2.py "10.1" --file="MyFile"
    python pronouns2.py --help
    
    Additional boolean options from Manager.state can be specified:
    --option_name          (sets option to True)
    --option_name=True/False
    
    Special options:
    --file_type=.txt/.tex (sets output file type)
    --chaining_rho=0.0-1.0 (sets chaining parameter)
    
    Returns:
        None
    """

    # Create parser
    parser = argparse.ArgumentParser(description='Generate analyses for examples')
    # Add positional argument for numbers (optional)
    parser.add_argument('numbers', nargs='?', default=None,
                       help='Example numbers to document (e.g. "10.1")')
    # Add --file argument
    parser.add_argument('--file', default="../docs/Demo",
                       help='Output file path (default: ../docs/Demo)')
    # Add arguments for all Manager.state keys (except 'stream')
    valid_state_keys = set(Manager.manager.state.keys()) - {'stream'}
    for key in valid_state_keys:
        if key == 'chaining_rho':
            parser.add_argument(f'--{key}', type=float,
                              help='Chaining parameter (0.0-1.0)')
        elif key == 'file_type':
            parser.add_argument(f'--{key}', choices=['.txt', '.tex'],
                              help='Output file type')
        else:
            parser.add_argument(f'--{key}', nargs='?', const=True, type=bool,
                              help=f'Set {key} (default: {Manager.manager.state[key]})')
    # Parse arguments
    args = parser.parse_args()
    # Build kwargs dictionary for doc()
    kwargs = {}
    # Handle special cases first
    if args.file_type:
        kwargs['file_type'] = FileType.from_ext(args.file_type)
    if args.chaining_rho is not None:
        if not 0.0 <= args.chaining_rho <= 1.0:
            parser.error("--chaining_rho must be between 0.0 and 1.0")
        kwargs['chaining_rho'] = args.chaining_rho
    # Add all other boolean options from valid_state_keys
    for key in valid_state_keys:
        if key not in {'chaining_rho', 'file_type'}:
            value = getattr(args, key)
            if value is not None:
                kwargs[key] = value
    # Call doc() with parsed arguments
    full_path = doc(numbers=args.numbers, file=args.file, **kwargs)
    print(f"Created: {full_path}")

if __name__ == '__main__':
    main()
