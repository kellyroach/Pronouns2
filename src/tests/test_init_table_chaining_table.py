################################################################
#
#    test_init_table_chaining_table.py
#
################################################################

from unittest import TestCase
from tests.utils.run_test_doc import *
from pronouns2 import *

def run_test_init_table_chaining_table(file: str, **kwargs) -> None:
    """Shared subroutine for testing init_table table_proc_write_chaining_table.

    Args:
        file (str): File name.
    """
    kwargs.update({'info': False,
                   'features_table': False,
                   'abstract_diagram': False,
                   'nodes_after_chaining': False,
                   'nodes_table': False,
                   'chaining_diagram': False,
                   'chaining_table': True,
                   'interpretations_table': False,
                   'summary_table': False,
                   'lexicon_table': False,
                   'debug': True,
                   'nodes_after_parse': False,
                   'init_table': True,
                   'new_chain': False,
                   'trace': False})
    run_test_doc(file=file, **kwargs)

class TestInitTableChainingTable(TestCase):
    def test_init_table_chaining_table_txt(self):
        """Test doc with file_type=FileType.TXT."""
        run_test_init_table_chaining_table(file="test_init_table_chaining_table.txt",
                                           file_type=FileType.TXT)
    def test_init_table_chaining_table_tex(self):
        """Test doc with file_type=FileType.TEX."""
        run_test_init_table_chaining_table(file="test_init_table_chaining_table.tex",
                                           file_type=FileType.TEX)
