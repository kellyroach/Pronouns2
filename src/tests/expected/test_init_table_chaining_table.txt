################################################################
#
#     test_init_table_chaining_table.txt
#
################################################################

################################################################
#
#     (1.1) The boy who was fooling her kissed the girl who loved him.
#
################################################################

___________CHAINING___________
!      !      !       !      !
! boy  ! her  ! girl  ! him  !
! boyA ! herA ! girlA ! himA !
!______!______!_______!______!

################################################################
#
#     (1.2) *John killed herself.
#
################################################################

______CHAINING______
!       !          !
! John  ! herself  !
! JohnA ! herselfA !
!_______!__________!

################################################################
#
#     (1.4) Some students think they are smarter than they are.
#
################################################################

_______CHAINING______
!           !       !
! students  ! they  !
! studentsA ! theyA !
!___________!_______!

################################################################
#
#     (1.6) My uncle has never ridden a camel but his brother has, although it was lame.
#
################################################################

_____________________CHAINING____________________
!     !        !        !      !          !     !
! my  ! uncle  ! camel  ! his  ! brother  ! it  !
! myA ! uncleA ! camelA ! hisA ! brotherA ! itA !
!_____!________!________!______!__________!_____!

################################################################
#
#     (1.10) I like the fresh candy better than the stale PHI.
#
################################################################

_______CHAINING_______
!    !        !      !
! I  ! candy  ! PHI  !
! IA ! candyA ! PHIA !
!____!________!______!

################################################################
#
#     (1.11) If John can, he will do it.
#
################################################################

____CHAINING___
!       !     !
! John  ! he  !
! JohnA ! heA !
!_______!_____!

################################################################
#
#     (1.12) If he can, John will do it.
#
################################################################

____CHAINING___
!     !       !
! he  ! John  !
! heA ! JohnA !
!_____!_______!

################################################################
#
#     (1.13) John will do it if he can.
#
################################################################

____CHAINING___
!       !     !
! John  ! he  !
! JohnA ! heA !
!_______!_____!

################################################################
#
#     (1.14) He will do it if John can.
#
################################################################

____CHAINING___
!     !       !
! he  ! John  !
! heA ! JohnA !
!_____!_______!

################################################################
#
#     (1.24) I have a cat at home, but hate it.
#
################################################################

_____________CHAINING_____________
!    !      !       !      !     !
! I  ! cat  ! home  ! PHI  ! it  !
! IA ! catA ! homeA ! PHIA ! itA !
!____!______!_______!______!_____!

################################################################
#
#     (1.25) I want to get a cat for myself.
#
################################################################

___________CHAINING___________
!    !      !      !         !
! I  ! PHI  ! cat  ! myself  !
! IA ! PHIA ! catA ! myselfA !
!____!______!______!_________!

################################################################
#
#     (1.27) The men took off their hats.
#
################################################################

_________CHAINING________
!      !        !       !
! men  ! their  ! hats  !
! menA ! theirA ! hatsA !
!______!________!_______!

################################################################
#
#     (1.51) The man who lives next door said that he would mow my lawn.
#
################################################################

__________CHAINING__________
!      !     !     !       !
! man  ! he  ! my  ! lawn  !
! manA ! heA ! myA ! lawnA !
!______!_____!_____!_______!

################################################################
#
#     (1.52) Somebody seduced Bill's sister, but no one will ever seduce Jack’s and she knows it.
#
################################################################

_________________________CHAINING________________________
!           !         !         !         !      !      !
! somebody  ! Bill's  ! sister  ! Jack's  ! PHI  ! she  !
! somebodyA ! Bill'sA ! sisterA ! Jack'sA ! PHIA ! sheA !
!___________!_________!_________!_________!______!______!

################################################################
#
#     (1.66) Realizing that he was unpopular didn't disturb Oscar.
#
################################################################

________CHAINING_______
!      !     !        !
! PHI  ! he  ! Oscar  !
! PHIA ! heA ! OscarA !
!______!_____!________!

################################################################
#
#     (1.70) My neighbor who is pregnant said that she was very happy.
#
################################################################

_________CHAINING_________
!     !           !      !
! my  ! neighbor  ! she  !
! myA ! neighborA ! sheA !
!_____!___________!______!

################################################################
#
#     (1.74) The pilot who shot at it hit the Mig that chased him.
#
################################################################

___________CHAINING___________
!        !     !      !      !
! pilot  ! it  ! Mig  ! him  !
! pilotA ! itA ! MigA ! himA !
!________!_____!______!______!

################################################################
#
#     (1.86) The mosquito which bit Algernon was killed by him.
#
################################################################

____________CHAINING____________
!           !           !      !
! mosquito  ! Algernon  ! him  !
! mosquitoA ! AlgernonA ! himA !
!___________!___________!______!

################################################################
#
#     (1.87) The mosquito which bit him was killed by Algernon.
#
################################################################

____________CHAINING____________
!           !      !           !
! mosquito  ! him  ! Algernon  !
! mosquitoA ! himA ! AlgernonA !
!___________!______!___________!

################################################################
#
#     (1.88) Algernon killed the mosquito which bit him.
#
################################################################

____________CHAINING____________
!           !           !      !
! Algernon  ! mosquito  ! him  !
! AlgernonA ! mosquitoA ! himA !
!___________!___________!______!

################################################################
#
#     (1.89) He killed the mosquito which bit Algernon.
#
################################################################

____________CHAINING___________
!     !           !           !
! he  ! mosquito  ! Algernon  !
! heA ! mosquitoA ! AlgernonA !
!_____!___________!___________!

################################################################
#
#     (1.91) After John Adams woke up, he was hungry.
#
################################################################

____CHAINING___
!       !     !
! John  ! he  !
! JohnA ! heA !
!_______!_____!

################################################################
#
#     (1.92) That Oscar was unpopular didn't disturb him.
#
################################################################

_____CHAINING____
!        !      !
! Oscar  ! him  !
! OscarA ! himA !
!________!______!

################################################################
#
#     (1.93) For your brother to refuse to pay taxes would get him into trouble.
#
################################################################

___________________CHAINING__________________
!       !          !      !      !          !
! your  ! brother  ! PHI  ! him  ! trouble  !
! yourA ! brotherA ! PHIA ! himA ! troubleA !
!_______!__________!______!______!__________!

################################################################
#
#     (1.94) Anna's complaining about Peter infuriated him.
#
################################################################

_________________CHAINING_________________
!         !              !        !      !
! Anna's  ! complaining  ! Peter  ! him  !
! Anna'sA ! complainingA ! PeterA ! himA !
!_________!______________!________!______!

################################################################
#
#     (1.95) The possibility that Fred will be unpopular doesn’t bother him.
#
################################################################

____________CHAINING___________
!              !       !      !
! possibility  ! Fred  ! him  !
! possibilityA ! FredA ! himA !
!______________!_______!______!

################################################################
#
#     (1.96) After he woke up, John Adams was hungry.
#
################################################################

____CHAINING___
!     !       !
! he  ! John  !
! heA ! JohnA !
!_____!_______!

################################################################
#
#     (1.97) That he was unpopular didn't disturb Oscar.
#
################################################################

____CHAINING____
!     !        !
! he  ! Oscar  !
! heA ! OscarA !
!_____!________!

################################################################
#
#     (1.98) For him to refuse to pay taxes would get your brother into trouble.
#
################################################################

_______________________CHAINING_______________________
!      !      !        !       !          !          !
! him  ! PHI  ! taxes  ! your  ! brother  ! trouble  !
! himA ! PHIA ! taxesA ! yourA ! brotherA ! troubleA !
!______!______!________!_______!__________!__________!

################################################################
#
#     (1.99) Anna's complaining about him infuriated Peter.
#
################################################################

_________________CHAINING_________________
!         !              !      !        !
! Anna's  ! complaining  ! him  ! Peter  !
! Anna'sA ! complainingA ! himA ! PeterA !
!_________!______________!______!________!

################################################################
#
#     (1.100) The possibility that he will be unpopular doesn’t bother Fred.
#
################################################################

___________CHAINING___________
!              !     !       !
! possibility  ! he  ! Fred  !
! possibilityA ! heA ! FredA !
!______________!_____!_______!

################################################################
#
#     (1.101) John Adams was hungry after he woke up.
#
################################################################

____CHAINING___
!       !     !
! John  ! he  !
! JohnA ! heA !
!_______!_____!

################################################################
#
#     (1.102) Oscar wasn't disturbed that he was unpopular.
#
################################################################

____CHAINING____
!        !     !
! Oscar  ! he  !
! OscarA ! heA !
!________!_____!

################################################################
#
#     (1.103) It would get your brother into trouble for him to refuse to pay taxes.
#
################################################################

_______________________CHAINING_______________________
!       !          !          !      !      !        !
! your  ! brother  ! trouble  ! him  ! PHI  ! taxes  !
! yourA ! brotherA ! troubleA ! himA ! PHIA ! taxesA !
!_______!__________!__________!______!______!________!

################################################################
#
#     (1.104) Peter was infuriated at Anna's complaining about him.
#
################################################################

_________________CHAINING_________________
!        !         !              !      !
! Peter  ! Anna's  ! complaining  ! him  !
! PeterA ! Anna'sA ! complainingA ! himA !
!________!_________!______________!______!

################################################################
#
#     (1.105) Fred isn't bothered by the possibility that he will be unpopular.
#
################################################################

___________CHAINING___________
!       !              !     !
! Fred  ! possibility  ! he  !
! FredA ! possibilityA ! heA !
!_______!______________!_____!

################################################################
#
#     (1.106) *He was hungry after John Adams woke up.
#
################################################################

____CHAINING___
!     !       !
! he  ! John  !
! heA ! JohnA !
!_____!_______!

################################################################
#
#     (1.107) *He wasn't disturbed that Oscar was unpopular.
#
################################################################

____CHAINING____
!     !        !
! he  ! Oscar  !
! heA ! OscarA !
!_____!________!

################################################################
#
#     (1.108) *It would get him into trouble for your brother to refuse to pay taxes.
#
################################################################

_______________________CHAINING_______________________
!      !          !       !          !      !        !
! him  ! trouble  ! your  ! brother  ! PHI  ! taxes  !
! himA ! troubleA ! yourA ! brotherA ! PHIA ! taxesA !
!______!__________!_______!__________!______!________!

################################################################
#
#     (1.109) *He was infuriated at Anna's complaining about Peter.
#
################################################################

_________________CHAINING________________
!     !         !              !        !
! he  ! Anna's  ! complaining  ! Peter  !
! heA ! Anna'sA ! complainingA ! PeterA !
!_____!_________!______________!________!

################################################################
#
#     (1.110) *He isn't bothered by the possibility that Fred will be unpopular.
#
################################################################

___________CHAINING___________
!     !              !       !
! he  ! possibility  ! Fred  !
! heA ! possibilityA ! FredA !
!_____!______________!_______!

################################################################
#
#     (1.111) Penelope cursed Peter and slandered him.
#
################################################################

______________CHAINING______________
!           !        !      !      !
! Penelope  ! Peter  ! PHI  ! him  !
! PenelopeA ! PeterA ! PHIA ! himA !
!___________!________!______!______!

################################################################
#
#     (1.112) *Penelope cursed him and slandered Peter.
#
################################################################

______________CHAINING______________
!           !      !      !        !
! Penelope  ! him  ! PHI  ! Peter  !
! PenelopeA ! himA ! PHIA ! PeterA !
!___________!______!______!________!

################################################################
#
#     (1.113) The interest in visiting Las Vegas that Mary displayed is typical of gamblers.
#
################################################################

___________________________CHAINING___________________________
!           !       !            !       !       !           !
! interest  ! PHI1  ! Las Vegas  ! Mary  ! PHI2  ! gamblers  !
! interestA ! PHI1A ! Las VegasA ! MaryA ! PHI2A ! gamblersA !
!___________!_______!____________!_______!_______!___________!

################################################################
#
#     (5.2) June hates flowers, but she waters them anyway.
#
################################################################

______________CHAINING_____________
!       !          !      !       !
! June  ! flowers  ! she  ! them  !
! JuneA ! flowersA ! sheA ! themA !
!_______!__________!______!_______!

################################################################
#
#     (8.1) John wants to give June a present, but he isn't sure she’ll like it.
#
################################################################

_______________________CHAINING_______________________
!       !      !       !          !     !      !     !
! John  ! PHI  ! June  ! present  ! he  ! she  ! it  !
! JohnA ! PHIA ! JuneA ! presentA ! heA ! sheA ! itA !
!_______!______!_______!__________!_____!______!_____!

################################################################
#
#     (10.1) John wants to give June a present, but he isn't sure she’ll like it.
#
################################################################

_______________________CHAINING_______________________
!       !      !       !          !     !      !     !
! John  ! PHI  ! June  ! present  ! he  ! she  ! it  !
! JohnA ! PHIA ! JuneA ! presentA ! heA ! sheA ! itA !
!_______!______!_______!__________!_____!______!_____!

################################################################
#
#     (10.2) Janet saw herself.
#
################################################################

_______CHAINING______
!        !          !
! Janet  ! herself  !
! JanetA ! herselfA !
!________!__________!

################################################################
#
#     (10.3) Janet saw her.
#
################################################################

_____CHAINING____
!        !      !
! Janet  ! her  !
! JanetA ! herA !
!________!______!

################################################################
#
#     (10.4) *Janet saw himself.
#
################################################################

_______CHAINING______
!        !          !
! Janet  ! himself  !
! JanetA ! himselfA !
!________!__________!

################################################################
#
#     (10.5) The men threw a smokescreen around themselves.
#
################################################################

_______________CHAINING______________
!      !              !             !
! men  ! smokescreen  ! themselves  !
! menA ! smokescreenA ! themselvesA !
!______!______________!_____________!

################################################################
#
#     (10.6) The men found a smokescreen around them.
#
################################################################

____________CHAINING___________
!      !              !       !
! men  ! smokescreen  ! them  !
! menA ! smokescreenA ! themA !
!______!______________!_______!

################################################################
#
#     (10.7) The men found a smokescreen to be around them.
#
################################################################

____________CHAINING___________
!      !              !       !
! men  ! smokescreen  ! them  !
! menA ! smokescreenA ! themA !
!______!______________!_______!

################################################################
#
#     (10.8) The men found a smokescreen and it was around them.
#
################################################################

_______________CHAINING______________
!      !              !     !       !
! men  ! smokescreen  ! it  ! them  !
! menA ! smokescreenA ! itA ! themA !
!______!______________!_____!_______!

################################################################
#
#     (10.9) I told John to protect himself.
#
################################################################

____________CHAINING____________
!    !       !      !          !
! I  ! John  ! PHI  ! himself  !
! IA ! JohnA ! PHIA ! himselfA !
!____!_______!______!__________!

################################################################
#
#     (10.10) I told John to protect me.
#
################################################################

__________CHAINING_________
!    !       !      !     !
! I  ! John  ! PHI  ! me  !
! IA ! JohnA ! PHIA ! meA !
!____!_______!______!_____!

################################################################
#
#     (10.11) I told John to protect myself.
#
################################################################

____________CHAINING___________
!    !       !      !         !
! I  ! John  ! PHI  ! myself  !
! IA ! JohnA ! PHIA ! myselfA !
!____!_______!______!_________!

################################################################
#
#     (11.12) Jack's house burned down, but he rebuilt it.
#
################################################################

____________CHAINING____________
!         !        !     !     !
! Jack's  ! house  ! he  ! it  !
! Jack'sA ! houseA ! heA ! itA !
!_________!________!_____!_____!

################################################################
#
#     (11.26) John owns some sheep and Harry vaccinates them.
#
################################################################

______________CHAINING_____________
!       !        !        !       !
! John  ! sheep  ! Harry  ! them  !
! JohnA ! sheepA ! HarryA ! themA !
!_______!________!________!_______!

################################################################
#
#     (11.27) Mary danced with many boys and they found her interesting.
#
################################################################

____________CHAINING____________
!       !       !       !      !
! Mary  ! boys  ! they  ! her  !
! MaryA ! boysA ! theyA ! herA !
!_______!_______!_______!______!

################################################################
#
#     (11.28) John lost a pen yesterday and Bill found one today.
#
################################################################

____________CHAINING___________
!       !      !       !      !
! John  ! pen  ! Bill  ! one  !
! JohnA ! penA ! BillA ! oneA !
!_______!______!_______!______!

################################################################
#
#     (11.29) John claimed to have found the solution to the problem, but Bill was sure he had found it.
#
################################################################

__________________________CHAINING_________________________
!       !      !           !          !       !     !     !
! John  ! PHI  ! solution  ! problem  ! Bill  ! he  ! it  !
! JohnA ! PHIA ! solutionA ! problemA ! BillA ! heA ! itA !
!_______!______!___________!__________!_______!_____!_____!

################################################################
#
#     (11.30) John wants to catch a fish and eat it for supper.
#
################################################################

_____________________CHAINING____________________
!       !       !       !       !     !         !
! John  ! PHI1  ! fish  ! PHI2  ! it  ! supper  !
! JohnA ! PHI1A ! fishA ! PHI2A ! itA ! supperA !
!_______!_______!_______!_______!_____!_________!

################################################################
#
#     (11.31) No one would put the blame on himself.
#
################################################################

______________CHAINING_____________
!      !      !        !          !
! one  ! PHI  ! blame  ! himself  !
! oneA ! PHIA ! blameA ! himselfA !
!______!______!________!__________!

################################################################
#
#     (11.32) Sue told Sandy about herself.
#
################################################################

__________CHAINING__________
!      !        !          !
! Sue  ! Sandy  ! herself  !
! SueA ! SandyA ! herselfA !
!______!________!__________!

################################################################
#
#     (11.33) *Jill kept talking about himself.
#
################################################################

______CHAINING______
!       !          !
! Jill  ! himself  !
! JillA ! himselfA !
!_______!__________!

################################################################
#
#     (11.34) Does Jack's making a pig of himself bother Bill?
#
################################################################

_______________CHAINING______________
!         !      !          !       !
! Jack's  ! pig  ! himself  ! Bill  !
! Jack'sA ! pigA ! himselfA ! BillA !
!_________!______!__________!_______!

################################################################
#
#     (11.35) John wants to give June a present, but he is afraid she won’t like it.
#
################################################################

_______________________CHAINING_______________________
!       !      !       !          !     !      !     !
! John  ! PHI  ! June  ! present  ! he  ! she  ! it  !
! JohnA ! PHIA ! JuneA ! presentA ! heA ! sheA ! itA !
!_______!______!_______!__________!_____!______!_____!

################################################################
#
#     (11.36) Ernie doesn't like Bernie, because he is such an asshole.
#
################################################################

_______________CHAINING______________
!        !         !     !          !
! Ernie  ! Bernie  ! he  ! asshole  !
! ErnieA ! BernieA ! heA ! assholeA !
!________!_________!_____!__________!

################################################################
#
#     (12.2) Mary's father killed himself.
#
################################################################

____________CHAINING____________
!         !         !          !
! Mary's  ! father  ! himself  !
! Mary'sA ! fatherA ! himselfA !
!_________!_________!__________!

################################################################
#
#     (12.3) *Mary's father killed him.
#
################################################################

__________CHAINING__________
!         !         !      !
! Mary's  ! father  ! him  !
! Mary'sA ! fatherA ! himA !
!_________!_________!______!

################################################################
#
#     (12.4) *Mary's father killed herself.
#
################################################################

____________CHAINING____________
!         !         !          !
! Mary's  ! father  ! herself  !
! Mary'sA ! fatherA ! herselfA !
!_________!_________!__________!

################################################################
#
#     (12.5) Mary's father killed her.
#
################################################################

__________CHAINING__________
!         !         !      !
! Mary's  ! father  ! her  !
! Mary'sA ! fatherA ! herA !
!_________!_________!______!

################################################################
#
#     (12.6) The father of Mary killed himself.
#
################################################################

___________CHAINING___________
!         !       !          !
! father  ! Mary  ! himself  !
! fatherA ! MaryA ! himselfA !
!_________!_______!__________!

################################################################
#
#     (12.7) *The father of Mary killed him.
#
################################################################

_________CHAINING_________
!         !       !      !
! father  ! Mary  ! him  !
! fatherA ! MaryA ! himA !
!_________!_______!______!

################################################################
#
#     (12.8) *The father of Mary killed herself.
#
################################################################

___________CHAINING___________
!         !       !          !
! father  ! Mary  ! herself  !
! fatherA ! MaryA ! herselfA !
!_________!_______!__________!

################################################################
#
#     (12.9) The father of Mary killed her.
#
################################################################

_________CHAINING_________
!         !       !      !
! father  ! Mary  ! her  !
! fatherA ! MaryA ! herA !
!_________!_______!______!

################################################################
#
#     (12.11) Mary's mother cooks only for herself.
#
################################################################

____________CHAINING____________
!         !         !          !
! Mary's  ! mother  ! herself  !
! Mary'sA ! motherA ! herselfA !
!_________!_________!__________!

################################################################
#
#     (12.12) Mary's mother cooks only for her.
#
################################################################

__________CHAINING__________
!         !         !      !
! Mary's  ! mother  ! her  !
! Mary'sA ! motherA ! herA !
!_________!_________!______!

################################################################
#
#     (12.13) Mary's mother cooks only for her mother.
#
################################################################

________________CHAINING________________
!         !          !      !          !
! Mary's  ! mother1  ! her  ! mother2  !
! Mary'sA ! mother1A ! herA ! mother2A !
!_________!__________!______!__________!

################################################################
#
#     (13.2) It was difficult to sketch myself.
#
################################################################

____________CHAINING____________
!     !       !      !         !
! I0  ! you0  ! PHI  ! myself  !
! I0A ! you0A ! PHIA ! myselfA !
!_____!_______!______!_________!

################################################################
#
#     (13.5a) It was difficult for me to sketch myself.
#
################################################################

____________CHAINING___________
!     !       !     !         !
! I0  ! you0  ! me  ! myself  !
! I0A ! you0A ! meA ! myselfA !
!_____!_______!_____!_________!

################################################################
#
#     (13.6a) It was difficult for you to sketch yourself.
#
################################################################

_____________CHAINING_____________
!     !       !      !           !
! I0  ! you0  ! you  ! yourself  !
! I0A ! you0A ! youA ! yourselfA !
!_____!_______!______!___________!

################################################################
#
#     (13.7a) It was difficult for him to sketch himself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! him  ! himself  !
! I0A ! you0A ! himA ! himselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.8a) It was difficult for her to sketch herself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! her  ! herself  !
! I0A ! you0A ! herA ! herselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.9a) It was difficult for us to sketch ourselves.
#
################################################################

_____________CHAINING_____________
!     !       !     !            !
! I0  ! you0  ! us  ! ourselves  !
! I0A ! you0A ! usA ! ourselvesA !
!_____!_______!_____!____________!

################################################################
#
#     (13.10a) It was difficult for you to sketch yourselves.
#
################################################################

______________CHAINING______________
!     !       !      !             !
! I0  ! you0  ! you  ! yourselves  !
! I0A ! you0A ! youA ! yourselvesA !
!_____!_______!______!_____________!

################################################################
#
#     (13.11a) It was difficult for them to sketch themselves.
#
################################################################

_______________CHAINING______________
!     !       !       !             !
! I0  ! you0  ! them  ! themselves  !
! I0A ! you0A ! themA ! themselvesA !
!_____!_______!_______!_____________!

################################################################
#
#     (13.5b) It was difficult to sketch myself.
#
################################################################

____________CHAINING____________
!     !       !      !         !
! I0  ! you0  ! PHI  ! myself  !
! I0A ! you0A ! PHIA ! myselfA !
!_____!_______!______!_________!

################################################################
#
#     (13.6b) It was difficult to sketch yourself.
#
################################################################

_____________CHAINING_____________
!     !       !      !           !
! I0  ! you0  ! PHI  ! yourself  !
! I0A ! you0A ! PHIA ! yourselfA !
!_____!_______!______!___________!

################################################################
#
#     (13.7b) *It was difficult to sketch himself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! PHI  ! himself  !
! I0A ! you0A ! PHIA ! himselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.8b) *It was difficult to sketch herself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! PHI  ! herself  !
! I0A ! you0A ! PHIA ! herselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.9b) It was difficult to sketch ourselves.
#
################################################################

______________CHAINING_____________
!     !       !      !            !
! I0  ! you0  ! PHI  ! ourselves  !
! I0A ! you0A ! PHIA ! ourselvesA !
!_____!_______!______!____________!

################################################################
#
#     (13.10b) It was difficult to sketch yourselves.
#
################################################################

______________CHAINING______________
!     !       !      !             !
! I0  ! you0  ! PHI  ! yourselves  !
! I0A ! you0A ! PHIA ! yourselvesA !
!_____!_______!______!_____________!

################################################################
#
#     (13.11b) *It was difficult to sketch themselves.
#
################################################################

______________CHAINING______________
!     !       !      !             !
! I0  ! you0  ! PHI  ! themselves  !
! I0A ! you0A ! PHIA ! themselvesA !
!_____!_______!______!_____________!

################################################################
#
#     (13.12a) Nurse Bob Breezy gave up drawing.
#
################################################################

_______CHAINING_______
!     !       !      !
! I0  ! you0  ! Bob  !
! I0A ! you0A ! BobA !
!_____!_______!______!

################################################################
#
#     (13.12b) [Bob] It was difficult to sketch himself.
#
################################################################

________________CHAINING________________
!     !       !      !      !          !
! I0  ! you0  ! Bob  ! PHI  ! himself  !
! I0A ! you0A ! BobA ! PHIA ! himselfA !
!_____!_______!______!______!__________!

################################################################
#
#     (13.13a) Astronaut Linda Smith gave up drawing.
#
################################################################

________CHAINING________
!     !       !        !
! I0  ! you0  ! Linda  !
! I0A ! you0A ! LindaA !
!_____!_______!________!

################################################################
#
#     (13.13b) [Linda] It was difficult to sketch herself.
#
################################################################

_________________CHAINING_________________
!     !       !        !      !          !
! I0  ! you0  ! Linda  ! PHI  ! herself  !
! I0A ! you0A ! LindaA ! PHIA ! herselfA !
!_____!_______!________!______!__________!

################################################################
#
#     (13.14a) The bank embezzlers gave up drawing.
#
################################################################

___________CHAINING__________
!     !       !             !
! I0  ! you0  ! embezzlers  !
! I0A ! you0A ! embezzlersA !
!_____!_______!_____________!

################################################################
#
#     (13.14b) [the bank embezzlers] It was difficult to sketch themselves.
#
################################################################

_____________________CHAINING_____________________
!     !       !             !      !             !
! I0  ! you0  ! embezzlers  ! PHI  ! themselves  !
! I0A ! you0A ! embezzlersA ! PHIA ! themselvesA !
!_____!_______!_____________!______!_____________!

################################################################
#
#     (13.15) *It was difficult to sketch himself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! PHI  ! himself  !
! I0A ! you0A ! PHIA ! himselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.16) *It was difficult to sketch herself.
#
################################################################

_____________CHAINING____________
!     !       !      !          !
! I0  ! you0  ! PHI  ! herself  !
! I0A ! you0A ! PHIA ! herselfA !
!_____!_______!______!__________!

################################################################
#
#     (13.17) *It was difficult to sketch themselves.
#
################################################################

______________CHAINING______________
!     !       !      !             !
! I0  ! you0  ! PHI  ! themselves  !
! I0A ! you0A ! PHIA ! themselvesA !
!_____!_______!______!_____________!

################################################################
#
#     (13.18) [Bob] It was difficult to sketch himself.
#
################################################################

________________CHAINING________________
!     !       !      !      !          !
! I0  ! you0  ! Bob  ! PHI  ! himself  !
! I0A ! you0A ! BobA ! PHIA ! himselfA !
!_____!_______!______!______!__________!

################################################################
#
#     (13.19) [Linda] It was difficult to sketch herself.
#
################################################################

_________________CHAINING_________________
!     !       !        !      !          !
! I0  ! you0  ! Linda  ! PHI  ! herself  !
! I0A ! you0A ! LindaA ! PHIA ! herselfA !
!_____!_______!________!______!__________!

################################################################
#
#     (13.20) [the bank embezzlers] It was difficult to sketch themselves.
#
################################################################

_____________________CHAINING_____________________
!     !       !             !      !             !
! I0  ! you0  ! embezzlers  ! PHI  ! themselves  !
! I0A ! you0A ! embezzlersA ! PHIA ! themselvesA !
!_____!_______!_____________!______!_____________!

################################################################
#
#     (13.21) It was difficult to sketch myself.
#
################################################################

____________CHAINING____________
!     !       !      !         !
! I0  ! you0  ! PHI  ! myself  !
! I0A ! you0A ! PHIA ! myselfA !
!_____!_______!______!_________!

################################################################
#
#     (13.22) [toy] Give me that!
#
################################################################

__________________CHAINING_________________
!     !       !      !      !     !       !
! I0  ! you0  ! toy  ! PHI  ! me  ! that  !
! I0A ! you0A ! toyA ! PHIA ! meA ! thatA !
!_____!_______!______!______!_____!_______!
